x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/tasks/snapshot/snapshot.ts

```diff
@@ -165,6 +165,28 @@ const removeAllFieldsAndResetTimestamp: string = minifyPainless(`
     // Set the entity.last_seen_timestamp field to the current time
     newDoc.entity.last_seen_timestamp = params.timestampNow;
 
+    // Reset entity.behaviors fields if they exist
+    if (ctx._source.entity?.behaviors != null) {
+      if (params.entityType == 'generic') {
+        newDoc.entity.behaviors = new HashMap();
```

Comment:
painless syntax sugar (you don't need to use it, I'm just spreading the word of it):

```suggestion
        newDoc.entity.behaviors = [:];
```

Reply:
Nice tip! 🔥 

Took me a while to find a reference, so posting for anyone else - https://www.elastic.co/docs/reference/scripting-languages/painless/painless-operators-reference